packages/broker/test/unit/plugins/storage/Storage.test.ts

```diff
@@ -0,0 +1,255 @@
+
+    afterEach(async () => {
+        await storage!.close()
+    })
+
+    describe('requestLast', () => {
+
+        it('happy path', (done) => {
+            storage = createMockStorage(createResultFactory({
+                buckets: [MOCK_BUCKET],
+                messageCounts: [MOCK_MESSAGE_COUNT],
+                messages: [MOCK_MESSAGE_2]
+            }))
+            const resultStream = storage.requestLast(MOCK_STREAM_ID, 0, 1)
+            const onData = jest.fn()
+            resultStream.on('data', onData)
+            resultStream.on('end', () => {
+                expect(onData).toBeCalledTimes(1)
+                expect(onData.mock.calls[0][0].serialize()).toEqual(MOCK_MESSAGE_2.payload)
+                done()
+            })
+        })
+
+        it('no messages', (done) => {
+            storage = createMockStorage(createResultFactory({
+                buckets: [],
+            }))
+            const resultStream = storage.requestLast(MOCK_STREAM_ID, 0, 1)
+            const onData = jest.fn()
+            resultStream.on('data', onData)
+            resultStream.on('end', () => {
+                expect(onData).not.toBeCalled()
+                done()
+            })
+        })
+
+        it('bucket query error', (done) => {
+            const expectedError = new Error('bucket-error')
+            storage = createMockStorage(createResultFactory({
+                buckets: expectedError
+            }))
+            const resultStream = storage.requestLast(MOCK_STREAM_ID, 0, 1)
+            resultStream.on('error', (e) => {
+                expect(e).toBe(expectedError)
+                done()
+            })
+        })
+
+        it('message count query error', (done) => {
+            const expectedError = new Error('message-count-error')
+            storage = createMockStorage(createResultFactory({
+                buckets: [MOCK_BUCKET],
+                messageCounts: expectedError
+            }))
+            const resultStream = storage.requestLast(MOCK_STREAM_ID, 0, 1)
+            resultStream.on('error', (e) => {
+                expect(e).toBe(expectedError)
+                done()
+            })
+        })
+
+        it('message query error', (done) => {
+            const expectedError = new Error('message-error')
+            storage = createMockStorage(createResultFactory({
+                buckets: [MOCK_BUCKET],
+                messageCounts: [MOCK_MESSAGE_COUNT],
+                messages: expectedError
+            }))
+            const resultStream = storage.requestLast(MOCK_STREAM_ID, 0, 1)
+            resultStream.on('error', (e) => {
+                expect(e).toBe(expectedError)
+                done()
+            })
+        })
+
+    })
+
+    describe.each([
+        [REQUEST_TYPE_FROM, true],
+        [REQUEST_TYPE_FROM, false],
+        [REQUEST_TYPE_RANGE, true],
+        [REQUEST_TYPE_RANGE, false],
+    ])('%s, publisher: %p', (requestType: string, isPublisher: boolean) => {
```

Comment:
Got a bit confused here while reading variable `isPublisher`. First thought was whether someone is gonna start publishing messages here or something. It got cleared up thought as I read more. Think it is fine as is, but wondering if it was more clear just to define the actual `publisherId` value in the describe table (even though changing it to be different streams is not technically needed)?

Reply:
Fixed